frontend/src/metabase/nav/components/LicenseTokenMissingBanner/useLicenseTokenMissingBanner.ts

```diff
@@ -0,0 +1,75 @@
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+
+import { useSetting, useUserSetting } from "metabase/common/hooks";
+import { isEEBuild } from "metabase/lib/utils";
+import type { TokenStatus } from "metabase-types/api";
+
+dayjs.extend(utc);
+
+const DAYS_BEFORE_REPEAT_BANNER = 14;
+const MAX_NUMBER_OF_DISMISSALS = 2;
+
+export const getCurrentUTCTimestamp = () => {
+  return dayjs.utc().toISOString();
+};
+
+export function shouldShowBanner({
+  tokenStatus,
+  lastDismissed,
+  isEEBuild,
+}: {
+  tokenStatus: TokenStatus | null;
+  lastDismissed: Array<string>;
+  isEEBuild: boolean;
+}) {
+  if (!isEEBuild) {
+    return false;
+  }
+  if (tokenStatus !== null) {
+    return false;
+  }
+
+  if (lastDismissed.length >= MAX_NUMBER_OF_DISMISSALS) {
+    return false;
+  }
+  if (lastDismissed.length === 0) {
+    return true;
+  }
+
+  if (lastDismissed.length === 1) {
+    const daysSinceLastDismissed = dayjs(getCurrentUTCTimestamp()).diff(
+      lastDismissed[0],
+      "days",
+    );
+
+    if (daysSinceLastDismissed < DAYS_BEFORE_REPEAT_BANNER) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+export function useLicenseTokenMissingBanner() {
+  const [lastDismissed, setLastDismissed] = useUserSetting(
+    "license-token-missing-banner-dismissal-timestamp",
+  );
```

Comment:
now that this is not a user setting, you should not use `useUserSetting` , you can use `useAdminSetting` or just `useSetting` with `useUpdateSettingMutation`

But, since none of these have that debounce behavior, you should be able to get rid of that nasty test mock

Reply:
Right! I've updated this hook and tests https://github.com/metabase/metabase/pull/59030/commits/bb5edeb6f6f7e13d887bdf76170d5d78ec63717b


Indeed, mocking of `debounce` is not needed anymore.